cli/options.go

```diff
@@ -0,0 +1,161 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/sirupsen/logrus"
+)
+
+// ProjectOptions group the command line options recommended for a Compose implementation
+type ProjectOptions struct {
+	ConfigPaths []string
+	Name        string
+}
+
+// SupportedFilenames define the supported Compose file names for auto-discovery (in preference order)
+var SupportedFilenames = []string{"compose.yaml", "compose.yml", "docker-compose.yml", "docker-compose.yaml"}
+
+// ProjectFromOptions load a compose project based on command line options
+func ProjectFromOptions(options *ProjectOptions) (*types.Project, error) {
+	configPaths, err := getConfigPathsFromOptions(options)
+	if err != nil {
+		return nil, err
+	}
+
+	configs, err := parseConfigs(configPaths)
+	if err != nil {
+		return nil, err
+	}
+
+	return loader.Load(types.ConfigDetails{
+		ConfigFiles: configs,
+		Environment: environment(),
+	}, func(opts *loader.Options) {
+		if options.Name != "" {
+			opts.Name = options.Name
+		} else if nameFromEnv, ok := os.LookupEnv("COMPOSE_PROJECT_NAME"); ok {
+			opts.Name = nameFromEnv
+		}
+	})
+}
+
+// getConfigPathsFromOptions retrieve the config files for project based on project options
+func getConfigPathsFromOptions(options *ProjectOptions) ([]string, error) {
+	paths := []string{}
+	pwd, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(options.ConfigPaths) != 0 {
+		for _, f := range options.ConfigPaths {
+			if f == "-" {
+				paths = append(paths, f)
+				continue
+			}
+			if !filepath.IsAbs(f) {
+				f = filepath.Join(pwd, f)
+			}
+			if _, err := os.Stat(f); err != nil {
+				return nil, err
+			}
+			paths = append(paths, f)
+		}
+		return paths, nil
+	}
+
+	sep := os.Getenv("COMPOSE_FILE_SEPARATOR")
+	if sep == "" {
+		sep = string(os.PathListSeparator)
+	}
+	f := os.Getenv("COMPOSE_FILE")
+	if f != "" {
+		return strings.Split(f, sep), nil
+	}
+
+	for {
+		candidates := []string{}
+		for _, n := range SupportedFilenames {
+			f := filepath.Join(pwd, n)
+			if _, err := os.Stat(f); err == nil {
+				candidates = append(candidates, f)
+			}
+		}
+		if len(candidates) > 0 {
+			winner := candidates[0]
+			if len(candidates) > 1 {
+				logrus.Warnf("Found multiple config files with supported names: %s", strings.Join(candidates, ", "))
+				logrus.Warnf("Using %s\n", winner)
+			}
+			return []string{winner}, nil
+		}
+		parent := filepath.Dir(pwd)
+		if parent == pwd {
+			return nil, fmt.Errorf("can't find a suitable configuration file in this directory or any parent. Are you in the right directory?")
```

Comment:
Since this is a library, we should make the errors easy to handle for consumers. I'd recommend adding an errdefs package and doing something like `errors.Wrap(errdefs.ErrNotFound, "unable to find config file")`

It's also best to avoid giving user advice in the error itself. Stick to what's wrong.